homeassistant/components/media_player/webostv.py

```diff
@@ -144,19 +157,30 @@ def update(self):
             self._muted = self._client.get_muted()
             self._volume = self._client.get_volume()
             self._current_source_id = self._client.get_input()
-
             self._source_list = {}
-            self._source_label_list = []
             self._app_list = {}
+
+            custom_sources = []
+            for source in self._customize.get('sources', {}):
+                app_id = WEBOS_APPS_SHORT.get(source, None)
+                if app_id:
+                    custom_sources.append(app_id)
+                else:
+                    custom_sources.append(source)
+
             for app in self._client.get_apps():
                 self._app_list[app['id']] = app
+                if app['id'] == self._current_source_id:
+                    self._current_source = app['title']
+                    self._source_list[app['title']] = app
```

Comment:
line 176 will catch it if it's in the custom sources but someone might not have the current source in the sources list. so instead of having a dropdown for source select with nothing selected I'm adding it to the list.